src/types/Species.ts

```diff
@@ -10,8 +10,25 @@ export type Species = {
   rare?: boolean;
   seedStorageBehavior?: 'Orthodox' | 'Recalcitrant' | 'Intermediate' | 'Unknown';
   problems?: SpeciesProblemElement[];
+  ecosystemTypes?: EcosystemType[];
 };
 
```

Comment:
A thought.. if this was an enumeration, we could refer to them by enum key in the map below.

Reply:
Not sure I understand what you're asking here. `EcosystemType` is an enumeration (below).

Reply:
enum EcosystemType {
  BOREAL = ‘Boreal forests/Taiga’,
 …
}
etc but you are right this is overkill.
Objective was to help catch incorrect strings..
Another option I can think of is inferring this type from the generated schema.
type EcosystemType = paths[‘components’].. 
or at a minimum add return type to ecosystemTypes() to be Record<string, EcosystemType> so compiler can catch string errors. Maybe that’s the best..

Reply:
Oops return type should be
 { label: string; value: EcosystemType }[]

Reply:
I'll explore this a bit. Thanks!

Reply:
Thanks, adding a return type to ecosystemTypes() should be enough

Reply:
Got it, yeah that does make things safer 👍